src/main/java/org/jenkinsci/plugins/workflow/actions/TaskInfoAction.java

```diff
@@ -0,0 +1,84 @@
+package org.jenkinsci.plugins.workflow.actions;
+
+import hudson.model.InvisibleAction;
+import hudson.model.Queue;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+/**
+ * Records information for a {@code node} block.
+ */
+public abstract class TaskInfoAction extends InvisibleAction implements PersistentAction {
```

Comment:
This name is really hard to decipher. What is a “task” in this context and why would you need “info” on it?
 Perhaps `QueueItemAction` would be clearer?